### PR TITLE
[release/1.3.8] Bump client version and update deps to address CVEs

### PR DESCRIPTION
**Pull request by zhenmichael**
## Description

- Follow up PR to consume server 0.1036.5002 patch release: https://github.com/microsoft/FluidFramework/pull/19959
- Update dependencies in server npm packages to address [CVE-2023-45857](https://github.com/advisories/GHSA-wf5p-g6vw-rhxx) and [CVE-2024-21484](https://github.com/advisories/GHSA-rh63-9qcf-83gf).

### CODE REVIEW

**Comment by zhenmichael** [comment]
@alexvy86 The build tools pipeline seems to be running into an issue with "fluid-repo-policy-check: not found". So far I've tried to cherry pick the relevant changes from this PR https://github.com/microsoft/FluidFramework/pull/17389. Just wanted to check if it looks ok to you so far.

File: packages/dds/shared-object-base/src/sharedObject.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
      * to variables to avoid complaints from TypeScript.
      */
     private setUpSampledTelemetryHelpers(): SampledTelemetryHelper[] {
-        assert(this.mc !== undefined && this.logger !== undefined, "this.mc and/or this.logger has not been set");
```
Comment (alexvy86) [inline packages/dds/shared-object-base/src/sharedObject.ts:133]: Was this one part of the cherry-pick to get the pipelines to work? I wouldn't expect the assert tagging to have happened as part of those changes, nor as part of this PR. If it did come from the cherry pick, might be fine but need to think more about it.

File: packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ export class ObjectForest extends SimpleDependee implements IEditableForest {
     }
 
     private beforeChange(): void {
-        assert(this.currentCursors.size === 0, "No cursors can be current when modifying forest");
+        assert(this.currentCursors.size === 0, 0x348 /* No cursors can be current when modifying forest */);
```
Comment (alexvy86) [inline packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts:96]: Another assert tag

File: packages/loader/container-loader/src/deltaQueue.ts [hunk 3]
```diff
@@ -114,9 +114,9 @@ export class DeltaQueue<T>
         if (this.anythingToProcess() && this.processingPromise === undefined) {
             // Use a resolved promise to start the processing on a separate stack.
             this.processingPromise = Promise.resolve().then(() => {
-                assert(this.processingPromise !== undefined, "reentrancy?");
+                assert(this.processingPromise !== undefined, 0x349 /* reentrancy? */);
```
Comment (alexvy86) [inline packages/loader/container-loader/src/deltaQueue.ts:117]: More assert tags in this file

File: packages/runtime/container-runtime/src/runningSummarizer.ts [hunk 4]
```diff
@@ -290,7 +290,7 @@ export class RunningSummarizer implements IDisposable {
      */
     public async lockedRefreshSummaryAckAction<T>(action: () => Promise<T>) {
         assert(this.refreshSummaryAckLock === undefined,
-            "Refresh Summary Ack - Caller is responsible for checking lock");
+            0x34c /* Refresh Summary Ack - Caller is responsible for checking lock */);
```
Comment (alexvy86) [inline packages/runtime/container-runtime/src/runningSummarizer.ts:293]: One more here

**Review by alexvy86 — COMMENTED** [review]
Overall it looks correct. The version updates to the packages we want to update for CVES are there, the updates to use the new patch of server packages is there... so yeah. Would like to understand better where the changes to assert tags came from, but other than that I haven't seen anything particularly weird.

**Comment by zhenmichael** [comment]
Thanks for pointing it out, I don't think the assertions were in the original PR I cherry picked, I'll double check to see where the changes came from.

**Comment by zhenmichael** [comment]
moved to this PR https://github.com/microsoft/FluidFramework/pull/20288

**Pull request closed without merging**